src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,51 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+class Deadline extends Task {
+    protected String by;
```

Comment:
Would it be good to use a more descriptive variable name here? Right now "by" is not very indicative of the variable's function in the class. See '[Use name to explain](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/intermediate/useNameExplain/)' from the textbook.